src/server/pfs/server/driver.go

```diff
@@ -765,28 +739,23 @@ func (d *driver) makeCommit(
 // If a complete commit provenance is passed in it just uses that.
 // It accepts an STM so that it can be used in a transaction and avoids an
 // inconsistent call to d.inspectCommit()
-func (d *driver) resolveCommitProvenance(stm col.STM, userCommitProvenance *pfs.CommitProvenance) (*pfs.CommitProvenance, error) {
+func (d *driver) resolveCommitProvenance(stm col.STM, userCommitProvenance *pfs.CommitProvenance) (*pfs.CommitProvenance, *pfs.CommitInfo, error) {
```

Comment:
This was already the case but it strikes me as a little bit weird that this mutates one of its arguments and then returns it rather than just mutating it and having the user use it? Is there a good reason to do that?

Reply:
I also do not like this, but I was afraid of how many bugs I might introduce if I changed this.  I don't doubt that this has caused bugs by being the way it is either, though.